### DISCUSSION TITLE
Jquery + JQuery UI

### DISCUSSION THREAD

**Discussion by arnoldmontiel01** [breezefront/community#259] [1. Help]
Hello

I need to port an old module which uses Jquery + Jquery-ui into the breeze theme, I did added them using the breeze_defult.xml file like this:

                <argument name="bundles" xsi:type="array">
                    <item name="default" xsi:type="array">
                        <item name="items" xsi:type="array">
                            <item name="jquery-3.6" xsi:type="string">Vendor_CustomScripts/js/jquery-3.6</item>
                            <item name="jquery-ui" xsi:type="string">Vendor_CustomScripts/js/jquery-ui</item>
                        </item>
                    </item>
                </argument>

but after I did that I started to get a lot of errors within the whole site like:

Uncaught TypeError: Cannot read properties of undefined (reading 'regex')
    at default2.min.js:1:38
    at default2.min.js:2:90

or

Uncaught TypeError: $.view is not a function
    at product.min.js:82:490
    at product.min.js:82:1401

What do I need to do in order to be able to use Jquery+JquertUI? 

Thank you in advance

**Comment by vovayatsyuk** [comment]
I'm not sure if it's possible without breaking other things. You can try to do that using a separate bundle and no-conflict mode:

```xml
<referenceBlock name="breeze.js">
    <arguments>
        <argument name="bundles" xsi:type="array">
            <item name="jquery" xsi:type="array">
                <item name="active" xsi:type="boolean">true</item>
                <item name="sort_order" xsi:type="number">1</item>
                <item name="items" xsi:type="array">
                    <item name="jquery" xsi:type="string">Vendor_CustomScripts/js/jquery</item>
                    <item name="jquery-no-conflict" xsi:type="string">Vendor_CustomScripts/js/jquery-no-conflict</item>
                    <item name="jquery-ui" xsi:type="string">Vendor_CustomScripts/js/jquery-ui</item>
                </item>
            </item>
        </argument>
    </arguments>
</referenceBlock>
```

Contents of jquery-no-conflict.js:

```js
$.noConflict();
```

Now, you can use `jQuery()`, while `$` still refers to the Cash. 

If you want to use `$`, follow this approach:

```js
(($) => {
    // $ - is jQuery here
})(jQuery);
```
